File: src/p002/p002.jl
function interativesolution()
    curr = 0
    nxt = 1
    total = 0

    while curr < 4_000_000
        if curr % 2 == 0
            total += curr
        end

        curr, nxt = nxt, curr + nxt
    end

    total
end

println(interativesolution())

File: src/p002/p002.lua
#!/usr/bin/env lua

local function iterativesolution()
  local curr = 0
  local nxt = 1
  local total = 0

  while curr < 4000000 do
    if curr % 2 == 0 then
      total = total + curr
    end

    curr, nxt = nxt, curr + nxt
  end

  return total
end

print(iterativesolution())
